mcs/class/System/Mono.Btls/MonoBtlsProvider.cs

```diff
@@ -201,21 +201,80 @@ internal static bool ValidateCertificate (MonoBtlsX509Chain chain, MonoBtlsX509V
 			}
 		}
 
+		internal static void SetupCertificateStore (MonoBtlsX509Store store, MonoTlsSettings settings, bool server)
+		{
+#if MONODROID
+				SetupCertificateStore (store);
+				return;
+#else
+
+			if (settings == null || settings.CertificateSearchPaths == null) {
+				SetupCertificateStore (store);
+				return;
+			}
+
+			foreach (var path in settings.CertificateSearchPaths) {
+				if (string.Equals (path, "@default", StringComparison.Ordinal)) {
+					AddTrustedRoots (store, settings, server);
+					AddUserStore (store);
+					AddMachineStore (store);
+				} else if (string.Equals (path, "@user", StringComparison.Ordinal))
+					AddUserStore (store);
+				else if (string.Equals (path, "@machine", StringComparison.Ordinal))
+					AddMachineStore (store);
+				else if (string.Equals (path, "@trusted", StringComparison.Ordinal))
+					AddTrustedRoots (store, settings, server);
+				else if (path.StartsWith ("@pem:", StringComparison.Ordinal)) {
+					var realPath = path.Substring (5);
+					if (Directory.Exists (realPath))
+						store.AddDirectoryLookup (realPath, MonoBtlsX509FileType.PEM);
+				} else if (path.StartsWith ("@der:", StringComparison.Ordinal)) {
+					var realPath = path.Substring (5);
+					if (Directory.Exists (realPath))
+						store.AddDirectoryLookup (realPath, MonoBtlsX509FileType.ASN1);
+				} else {
+					if (Directory.Exists (path))
```

Comment:
Ok, I thought about this a bit more and decided against using the emum and that accessor method as having an "add"-type method wouldn't make any sense - we'd either need a method taking a list or no method at all.

The reasoning is that you are not "adding" things - the moment you assign anything to this property, you replace the default search paths (and need to explicitly add "@default" if you still want them).

I'm going to make the property internal, though - so we don't need to commit on a public API yet.